tests/rules/metavar_type_invalid.yaml

```diff
@@ -0,0 +1,11 @@
+rules:
+  - id: test
+    message: Test
+    languages:
+      - java
+    severity: WARNING
+    patterns:
+      - pattern: private $T x;
+      - metavariable-type:
+          metavariable: $T
```

Comment:
why this is not matching??? Or did you make a typo and you wanted to write 'metavariable: $X' (which is not bounded in the pattern, like you had in the changelog).

Reply:
> why this is not matching???

`$T` represents a type identifier so it doesn't match with a type itself. `private int $T` would match.

And I think Iago also wanted to write `metavariable: $X`.

Reply:
Ooops, sorry, I wrote the wrong test indeed.